scipy/optimize/_milp.py

```diff
@@ -0,0 +1,280 @@
+import numpy as np
+from scipy.sparse import csc_array
+from ._highs._highs_wrapper import _highs_wrapper
+from ._constraints import LinearConstraint, Bounds
+from ._optimize import OptimizeResult
+
+
+def _milp_iv(c, integrality, bounds, constraints, options):
+
+    c = np.atleast_1d(c).astype(np.double)
+    if c.ndim != 1:
+        message = "`c` must be a one-dimensional array."
+        raise ValueError(message)
+
+    if integrality is None:
+        integrality = 0
+    integrality = np.broadcast_to(integrality, c.shape).astype(np.uint8)
+
+    if bounds is None:
+        bounds = Bounds(0, np.inf)
+    elif not isinstance(bounds, Bounds):
+        message = "`bounds` must be an instance of `scipy.optimize.Bounds`."
+        try:
+            bounds = Bounds(*bounds)
+        except TypeError:
+            raise ValueError(message)
+    # broadcast in case lb/ub are scalars
+    lb = np.broadcast_to(bounds.lb, c.shape).astype(np.double)
+    ub = np.broadcast_to(bounds.ub, c.shape).astype(np.double)
+
+    if constraints is None:
+        constraints = LinearConstraint(np.empty((0, c.size)),
+                                       np.empty((0,)), np.empty((0,)))
+    elif not isinstance(constraints, LinearConstraint):
+        message = ("`constraints` must be an instance of "
+                   "`scipy.optimize.LinearConstraint`.")
+        try:
+            constraints = LinearConstraint(*constraints)
```

Comment:
Do we understand the overhead in using `Bounds` and `LinearConstraint` objects over linprog-style matrices/tuple-lists?  I remember when investigating other LP/MIP Python libraries that model construction could have significant overhead in terms of actual time-to-solution

Reply:
They are merely containers; essentially zero overhead.

Reply:
Essentially done with improvements to those classes. See gh-15297 and mdhaber/scipy#67. 
Now, in addition to accepting inputs of type `Bounds`/`LinearConstraint`, I also accept tuples that can be converted to those types like `Bounds(*bounds)`